src/components/ProjectDomain/projectdomainTable.tsx

```diff
@@ -0,0 +1,302 @@
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await apiHelperService.getAllProjectdomain();
+      if (!response.data.data) {
+        setNoData(true); // Set noData if response is empty
+      } else {
+        setDomainData(response.data.data);
+      }
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Failed to fetch domain data. Please Refresh the page.",
+        variant: "destructive", // Red error message
+      });
+
+      setNoData(true); // Handle errors by showing no data
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Fetch domain data on component mount
+  useEffect(() => {
+    fetchDomainData();
+  }, []);
+
+  // Handle domain deletion
+  const handleDelete = async (domainId: string) => {
+    if (!domainId) {
+      toast({
+        title: "Error",
+        description: "Failed there is no such id . Please try again.",
+        variant: "destructive", // Red error message
+      });
+      return;
+    }
+    try {
+      await apiHelperService.deleteProjectdomain(domainId);
+      fetchDomainData(); // Re-fetch data after deletion
+    } catch (error: any) {
+      toast({
+        title: "Error",
+        description: "Failed to delete domain . Please try again.",
+        variant: "destructive", // Red error message
+      });
+    }
+  };
+
+  const handleSwitchChange = async (labelId: string, checked: boolean) => {
+    // Initialize toast
+    setDomainData((prevData) =>
+      prevData.map((user) =>
+        user._id === labelId
+          ? {
+              ...user,
+              status: checked ? statusType.active : statusType.inactive,
+            }
+          : user,
+      ),
+    );
+    try {
+      await axiosInstance.put(`/projectdomain/${labelId}`, {
+        status: checked ? statusType.active : statusType.inactive,
+      });
+      toast({
+        title: "Success",
+        description: `Domain status updated to ${checked ? statusType.active : statusType.inactive}`,
+        variant: "default",
+      });
+    } catch (error) {
+      // Revert the status change if the API call fails
+      setDomainData((prevData) =>
+        prevData.map((domain) =>
+          domain._id === labelId
+            ? {
+                ...domain,
+                status: checked ? statusType.inactive : statusType.active,
+              } // revert back to original status
+            : domain,
+        ),
+      );
+      toast({
+        title: "Error",
+        description: "Failed to update domain status. Please try again.",
+        variant: "destructive", // Red error message
+      });
+    }
+  };
+  const formatID = (id: string) => {
+    if (id.length <= 7) return id;
+    return `${id.substring(0, 5)}...${id.substring(id.length - 2)}`;
+  };
+  // Callback to re-fetch domain data after adding a new domain
+  const handleAddDomain = async () => {
+    try {
+      // Optionally post newDomain if needed
```

Comment:
why is this function named handleAddDomain and it is just calling fetchDomainData()

Reply:
This function is not required if its job is to just call fetchDomainData() function, you can pass the function fetchDomainData() directly to onAddProjectDomain prop

Reply:
this function is created so that is the new domain is added it should be reflected to the main table without reloading